compiler/src/lowering/mappers/functions.rs

```diff
@@ -180,3 +184,85 @@ fn is_last_statement_return(stmts: &[Node<fe::FuncStmt>]) -> bool {
         false
     }
 }
+
+/// Lowers tuple desctructuring.
+///
+/// e.g.
+/// ```fe
+/// (x, y): (uint256, bool) = (1, true)
+/// ```
+/// will be lowered to
+/// ```fe
+/// $tmp_tuple_0: (uint256, bool) = (1, true)
+/// x: uint256 = $tmp_tuple_0.item0
+/// y: bool = $tmp_tuple_0.item1
+/// ```
```

Comment:
Also, I suppose `(uint256, bool)` becomes `tuple_u256_bool` with the current tuple type lowering,

Reply:
Yeah, I saw the lowering tests and tried to add a test. But I think we can't pass the test with the current renaming strategy because the parser rejects `$`, so the test will fail in [here](https://github.com/ethereum/fe/blob/9686ec3641124139f6edd05e41da92eecc449a7c/tests/src/lowering/mod.rs#L70).

Reply:
Ah, good point,